nav2_waypoint_follower/src/waypoint_follower.cpp

```diff
@@ -304,4 +379,58 @@ WaypointFollower::goalResponseCallback(
   }
 }
 
+std::vector<geometry_msgs::msg::PoseStamped>
+WaypointFollower::convertGPSPoses2MapPoses(
+  const std::vector<nav2_msgs::msg::OrientedNavSatFix> & gps_poses,
+  const rclcpp_lifecycle::LifecycleNode::SharedPtr & parent_node,
+  const std::unique_ptr<nav2_util::ServiceClient<robot_localization::srv::FromLL>> & fromll_client)
+{
+  RCLCPP_INFO(parent_node->get_logger(), "Converting GPS waypoints to Map Frame..");
+  rclcpp::Duration transform_tolerance(0, 500);
+
+  std::vector<geometry_msgs::msg::PoseStamped> poses_in_map_frame_vector;
+  for (auto && curr_gps_pose : gps_poses) {
+    auto request = std::make_shared<robot_localization::srv::FromLL::Request>();
+    auto response = std::make_shared<robot_localization::srv::FromLL::Response>();
+    request->ll_point.latitude = curr_gps_pose.position.latitude;
+    request->ll_point.longitude = curr_gps_pose.position.longitude;
+    request->ll_point.altitude = curr_gps_pose.position.altitude;
+
+    fromll_client->wait_for_service((std::chrono::seconds(1)));
+    auto is_conversion_succeeded = fromll_client->invoke(
+      request,
+      response);
+    if (!is_conversion_succeeded) {
+      RCLCPP_ERROR(
```

Comment:
More of a motivational question than a specific request: what do you think is the right behavior if some subset of the waypoints aren't able to be converted? Would we rather drop them or reject the request as failed? This seems like it could cause unintended consequences but I'd like to hear your thoughts.

Reply:
right now we are acting opportunistic, if conversion somehow fails then we drop that. To my understanding, there isn't one right answer to this but in my application I had rather have 7/10 of points converted rather than 0/10 out of 10 points. It is very likely that if conversion of one fails the rest will also fail and vice versa.

Reply:
True - but isn't that a critical error state if you need your robot to follow a particular path. A->B->C != A->C. I would think expected behavior would be to fail the entire action if the inputs cannot be verified or drivable. This would let the robot continue even though 1 or more of the inputs aren't drivable.

Reply:
For some applications where you want robot to strictly follow given set of waypoints this can be critical, my rationale is that, I trust obstacle information embedded to costmaps, so if something is on the way when robot jumps to `C ` without visiting `B`, I would still accomplish the task partially. One thing we can do here is; 
We can generalize `stop_on_failure` param so that it will work for both GPS and non-GPS. I suggest we name this as; 
`failure_of_one_waypoint_failure_of_all`, or something better, from there I can adjust the logic to comply with `stop_on_failure`

Reply:
> We can generalize stop_on_failure param so that it will work for both GPS and non-GPS.

Yes, good point, lets do that. I think just keep the variable name the same, that's still a `failure` to `stop_on_failure`, that parameter doesn't make specific mention of what kind of failure, its all kinds of failure. The aim is so that people that are sensitive to EXACT following can exit as failed if anything goes wrong, but people that generally just need a robust system and "shit happens" need to be able to continue.